pkg/services/ngalert/notifier/alertmanager_config.go

```diff
@@ -155,40 +156,86 @@ func (moa *MultiOrgAlertmanager) gettableUserConfigFromAMConfigString(ctx contex
 	// Get the last known working configuration
-	_, err := moa.configStore.GetLatestAlertmanagerConfiguration(ctx, org)
+	_, err := moa.configStore.GetLatestAlertmanagerConfiguration(ctx, orgId)
 	if err != nil {
 		// If we don't have a configuration there's nothing for us to know and we should just continue saving the new one
 		if !errors.Is(err, store.ErrNoAlertmanagerConfiguration) {
 			return fmt.Errorf("failed to get latest configuration %w", err)
 		}
 	}
 
-	if err := moa.Crypto.ProcessSecureSettings(ctx, org, config.AlertmanagerConfig.Receivers); err != nil {
+	if err := moa.Crypto.ProcessSecureSettings(ctx, orgId, config.AlertmanagerConfig.Receivers); err != nil {
 		return fmt.Errorf("failed to post process Alertmanager configuration: %w", err)
 	}
 
 	if err := assignReceiverConfigsUIDs(config.AlertmanagerConfig.Receivers); err != nil {
 		return fmt.Errorf("failed to assign missing uids: %w", err)
 	}
 
-	am, err := moa.AlertmanagerFor(org)
+	am, err := moa.AlertmanagerFor(orgId)
 	if err != nil {
 		// It's okay if the alertmanager isn't ready yet, we're changing its config anyway.
 		if !errors.Is(err, ErrAlertmanagerNotReady) {
 			return err
 		}
 	}
 
-	if err := am.SaveAndApplyConfig(ctx, &config); err != nil {
+	if err := moa.SaveAndApplyConfig(ctx, orgId, am, &config); err != nil {
 		moa.logger.Error("Unable to save and apply alertmanager configuration", "error", err)
 		return AlertmanagerConfigRejectedError{err}
 	}
 
 	return nil
 }
 
+// SaveAndApplyConfig saves the given configuration to the database and applies it to the given alertmanager in a single transaction.
+func (moa *MultiOrgAlertmanager) SaveAndApplyConfig(ctx context.Context, orgID int64, am Alertmanager, cfg *definitions.PostableUserConfig) error {
+	rawConfig, err := json.Marshal(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to serialize to the Alertmanager configuration: %w", err)
+	}
+
+	cmd, err := createSaveAlertmanagerConfigurationCmd(orgID, string(rawConfig), false)
+	if err != nil {
+		return err
+	}
+
+	return moa.configStore.InTransaction(ctx, func(ctx context.Context) error {
+		return moa.configStore.SaveAlertmanagerConfigurationWithCallback(ctx, cmd, func(dbCfg models.AlertConfiguration) error {
+			return am.ApplyConfig(ctx, &dbCfg)
+		})
+	})
+}
+
+// SaveAndApplyDefaultConfig saves the default configuration to the database and applies it to the given alertmanager in a single transaction.
+func (moa *MultiOrgAlertmanager) SaveAndApplyDefaultConfig(ctx context.Context, orgID int64, am Alertmanager) error {
+	cmd, err := createSaveAlertmanagerConfigurationCmd(orgID, moa.settings.UnifiedAlerting.DefaultConfiguration, true)
+	if err != nil {
+		return err
+	}
+
+	return moa.configStore.InTransaction(ctx, func(ctx context.Context) error {
+		return moa.configStore.SaveAlertmanagerConfigurationWithCallback(ctx, cmd, func(dbCfg models.AlertConfiguration) error {
+			return am.ApplyConfig(ctx, &dbCfg)
+		})
+	})
+}
```

Comment:
Can we extract common parts from these methods into a new private method and make these call it?